File: app/src/main/java/ssilvalucas/testeandroidv2/data/model/LoginResponse.java
package ssilvalucas.testeandroidv2.data.model;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

public class LoginResponse implements Parcelable {

    @SerializedName("userAccount")
    private UserAccount userAccount;

    @SerializedName("error")
    private Error error;

    protected LoginResponse(Parcel parcel) {
        this.userAccount = parcel.readParcelable(UserAccount.class.getClassLoader());
        this.error       = parcel.readParcelable(Error.class.getClassLoader());
    }

    public static final Creator<LoginResponse> CREATOR = new Creator<LoginResponse>() {
        @Override
        public LoginResponse createFromParcel(Parcel in) {
            return new LoginResponse(in);
        }

        @Override
        public LoginResponse[] newArray(int size) {
            return new LoginResponse[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeParcelable(userAccount, flags);
        dest.writeParcelable(error, flags);
    }

    public UserAccount getUserAccount() {
        return userAccount;
    }

    public Error getError() {
        return error;
    }
}

File: ProjetoTesteSantanderK2/app/src/main/java/com/br/projetotestesantanter/api/model/LoginResponse.kt
package com.br.projetotestesantanter.api.model

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.SerializedName

data class LoginResponse(

    @SerializedName("userAccount")
    var userAccount : UserAccount,

    @SerializedName("error")
    var error: Error
): Parcelable{
    constructor(parcel: Parcel) : this(
        parcel.readParcelable(UserAccount::class.java.classLoader),
        parcel.readParcelable(Error::class.java.classLoader)
    )

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeParcelable(userAccount, flags)
        parcel.writeParcelable(error, flags)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<LoginResponse> {
        override fun createFromParcel(parcel: Parcel): LoginResponse {
            return LoginResponse(parcel)
        }

        override fun newArray(size: Int): Array<LoginResponse?> {
            return arrayOfNulls(size)
        }
    }

}
